storage/s3.go

```diff
@@ -1160,6 +1160,71 @@ func (s *S3) GetBucketVersioning(ctx context.Context, bucket string) (string, er
 
 }
 
+func (s *S3) HeadBucket(ctx context.Context, url *url.URL) (*Bucket, error) {
+
+	output, err := s.api.HeadBucketWithContext(ctx, &s3.HeadBucketInput{
+		Bucket: aws.String(url.Bucket),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if output == nil {
+		return nil, errors.New("output is nil")
+	}
```

Comment:
you can remove this. there is an unspoken agreement among go developers/libraries to me _-so take it with a grain of salt :salt:-_ ;

if a function returns a struct pointer and an error, unless otherwise stated, if the error is nil, the struct pointer is not nil. you can think it as  tagged union of value and an error.